src/environment.rs

```diff
@@ -322,6 +323,18 @@ pub fn lock_file_up_to_date(project: &Project, lock_file: &CondaLock) -> miette:
     Ok(true)
 }
 
+fn check_channel_package_url(channel: &str, url: &str) -> bool {
+    // Try to parse the channel string into a Channel type
+    // If this fails, the error will be propagated using `?`
+    let Ok(channel) = Channel::from_str(channel, &Default::default()) else {
+        return false;
+    };
+
+    // Check if the URL starts with the channel's base URL
+    // Return true or false accordingly
+    url.starts_with(channel.base_url.as_str())
```

Comment:
Is starts_with correct? Because it would still do the wrong thing for pytorch-nightly, no?

Reply:
See next comment